armi/reactor/blueprints/componentBlueprint.py

```diff
@@ -307,6 +320,46 @@ class ComponentKeyedList(yamlize.KeyedList):
     key_attr = ComponentBlueprint.name
 
 
+class GroupedComponent(yamlize.Object):
+    """
+    A pointer to a component with a multiplicity to be used in a ComponentGroup.
+
+    Multiplicity can be a fraction (e.g. to set volume fractions)
+    """
+
+    name = yamlize.Attribute(type=str)
+    mult = yamlize.Attribute(type=float)
+
+
+class ComponentGroup(yamlize.KeyedList):
+    """
+    A single component group containing multiple GroupedComponents
+
+    Example
+    -------
+    triso:
+      kernel:
+        mult: 0.7
+      buffer:
+        mult: 0.3
```

Comment:
This feels a little strange and clunky. The name `mult` in this case is more like a multiplier than a multiplicity (number of things). I guess having 0.7 `kernel` is kind of right... 

Secondly, this in theory would mean a user would need to convert the dimension properties to volume fractions. The math is easy but it's another possible step that could go wrong

Reply:
agreed. we need to make it more in line with your particlefuel class. I think we'll push this through as a stepping stone and then do #540 and then implement something that's basically identical to what you did, but with more first class support/usage of the builtin classes.